backend/src/config.rs

```diff
@@ -0,0 +1,159 @@
+use std::{fs::File, io::{Read, Write}, sync::{Mutex, Arc}, collections::HashMap, hash::Hash};
+use core::result::Result::Ok;
+use anyhow::{Result};
+use chrono::DateTime;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub enum ScrapTargetType {
+    User,
+    Repo,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub struct ScrapTarget {
+    pub username: String,
+    #[serde(skip_serializing_if = "Option::is_none", default)]
+    pub repo_names: Option<Vec<String>>,
+    pub target_type: ScrapTargetType,
+    
+    // ignore scrap target without removing them from config file
+    pub ignore: bool,
+}
+impl ScrapTarget {
+    pub fn user(username: String) -> Self {
+        Self {
+            username,
+            repo_names: None,
+            target_type: ScrapTargetType::User,
+            ignore: false,
+        }
+    }
+    pub fn repos(username: String, repo_names: Vec<String>) -> Self {
+        Self {
+            username,
+            repo_names: Some(repo_names),
+            target_type: ScrapTargetType::Repo,
+            ignore: false
+        }
+    }
+    
+    pub fn ignore(mut self) -> Self {
+        self.ignore = true;
+        self
+    }
+    pub fn target_links(&self) -> Vec<String> {
+        self.repo_names.as_ref()
+            .unwrap_or(&Vec::with_capacity(8))
```

Comment:
kenapa 8?

Reply:
Oh yea, no need to have extra allocation here